### ISSUE TITLE
500 | Internal Server Error when importing iCal file

### ISSUE THREAD

**Issue by pyrog** [betaglop/e-venement#9]
`SQLSTATE[22007]: Invalid datetime format: 7 ERREUR: syntaxe en entrée invalide pour le type timestamp : « 2016-2-13-21-00-00 0:00`

```sql
INSERT INTO manifestation
(automatic, online_limit, no_print, blocking, reservation_optional, reservation_confirmed, voucherized, event_id, happens_at, duration, location_id, vat_id, sf_guard_user_id, updated_at, reservation_begins_at, reservation_ends_at, created_at, version)
VALUES
(?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)',
array('false', 10, 'false', 'true', 'false', 'true', 'false', '4',
  '2016-2-13-21-00-00 0:00',
  0, '1', 1, 1,
  '2016-02-12 14:48:37',
  '2016-2-13-21-00-00 0:00',
  '1970-01-01 01:00:00',
  '2016-02-12 14:48:37'
  ), 
1
```

[star wars.ics.txt](https://github.com/betaglop/e-venement/files/128078/star.wars.ics.txt)

**Comment by betaglop** [comment]
Hi pyrog.

What is your environment (OS, PHP, DB...) ? What were you doing when this occurred?
It is really strange that you get this kind of date:  '2016-2-13-21-00-00 0:00'
It looks like nothing.

**Comment by betaglop** [comment]
sorry this occurs when you are importing an iCal file... it was given in the subject of your issue.
can you please check out your iCal file to find the datetimes as they are set inside? If necessary, you can post it here, will have a look into it.

**Comment by pyrog** [comment]
> What is your environment (OS, PHP, DB...) ?

* Ubuntu 15.10 - wily
* PHP Version 5.6.11-1ubuntu3.1
* PostgreSQL(libpq) Version 	9.4.5

> It is really strange that you get this kind of date: '2016-2-13-21-00-00 0:00'
> It looks like nothing.

The event start at 21:00. The good date in the db should be '2016-2-13 21:00'

> can you please check out your iCal file to find the datetimes as they are set inside?

The link to the file is at the bottom of the post :wink: 
**star wars.ics.txt**

Here, this is an excerpt for the first event:
``` vcal
BEGIN:VEVENT
CREATED:20160212T133847Z
UID:4334E77E-E157-47E5-8B0F-74D2B24486F7
DTEND;TZID=Europe/Paris:20160213T220000
TRANSP:OPAQUE
X-APPLE-TRAVEL-ADVISORY-BEHAVIOR:AUTOMATIC
SUMMARY:Star wars
DTSTART;TZID=Europe/Paris:20160213T210000
DTSTAMP:20160212T134022Z
LOCATION:Saint-Jean-du-Gard
SEQUENCE:0
END:VEVENT
```

DTSTART and DTEND are conform to the [rfc 5545](https://tools.ietf.org/html/rfc5545).
Example taken from the RFC: 
```
DTSTART;TZID=America/New_York:19970714T133000
                                                 ; Local time and time
                                                 ; zone reference

```

Also, this file is validated by [iCalendar Validator](http://severinghaus.org/projects/icv/?url=https%3A%2F%2Fgithub.com%2Fbetaglop%2Fe-venement%2Ffiles%2F128078%2Fstar.wars.ics.txt) (Source: [wikidot](http://icalvalid.wikidot.com/))

**Comment by betaglop** [comment]
I really don't know what happened.
Can you please upload your file or tell me again where I can found it (I did not find it reading again your last post)

**Comment by betaglop** [comment]
corrected in v2.10 with commit 333ef86
backported to v2.9 with commit 4393ed4
